cloud/scope/cluster.go

```diff
@@ -156,6 +157,13 @@ func (s *ClusterScope) RouteTableSpecs() []azure.RouteTableSpec {
 	}}
 }
 
+// NatGatewaySpecs returns the node nat gateway
+func (s *ClusterScope) NatGatewaySpecs() []azure.NatGatewaySpec {
+	return []azure.NatGatewaySpec{{
+		Name: s.NatGateway().Name,
```

Comment:
what if NatGateway is nil? we should only create it if explicitly specified by the user, ie. default behavior does not change